util/admin/admin.go

```diff
@@ -290,13 +293,16 @@ type RecordData struct {
 	Values []types.Datum
 }
 
-func getCount(ctx sessionctx.Context, sql string) (int64, error) {
-	rows, _, err := ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQLWithSnapshot(sql)
+func getCount(exec sqlexec.RestrictedSQLExecutor, stmt ast.StmtNode, snapshot uint64) (int64, error) {
+	rows, _, err := exec.ExecRestrictedStmt(context.Background(), stmt, sqlexec.ExecOptionWithSnapshot(snapshot))
 	if err != nil {
 		return 0, errors.Trace(err)
 	}
 	if len(rows) != 1 {
-		return 0, errors.Errorf("can not get count, sql %s result rows %d", sql, len(rows))
+		var sb strings.Builder
+		err1 := stmt.Restore(format.NewRestoreCtx(format.DefaultRestoreFlags, &sb))
+		terror.Log(err1)
+		return 0, errors.Errorf("can not get count, sql %s result rows %d", sb.String(), len(rows))
```

Comment:
Will this error propagated to the log so that internal sqls are exposed to users?

Reply:
The SQL string is removed